js/src/debugger/Source.cpp

```diff
@@ -202,6 +203,42 @@ bool DebuggerSource::CallData::ToNative(JSContext* cx, unsigned argc,
   return (data.*MyMethod)();
 }
 
+class DebuggerSourceGetHashMatcher {
+  JSContext* cx_;
+
+ public:
+  explicit DebuggerSourceGetHashMatcher(JSContext* cx) : cx_(cx) {}
+
+  using ReturnType = JSString*;
+
+  ReturnType match(HandleScriptSourceObject sourceObject) {
+    ScriptSource* ss = sourceObject->source();
+    bool hasSourceText;
+    if (!ScriptSource::loadSource(cx_, ss, &hasSourceText)) {
+      return nullptr;
+    }
+    if (!hasSourceText) {
+      return NewStringCopyZ<CanGC>(cx_, "[no source]");
```

Comment:
Seems like we should return `nullptr` rather than something that isn't a hash.

Reply:
nullptr is hard to distinguish from a failure-to-allocate.  I could return the empty string I guess?  Or wrap the return value in a `js::Maybe`

Reply:
Returning empty string and having the JS logic skip the call to upload the sourcemap is fine then.

Reply:
Haven't done it yet, but will do.